### ISSUE TITLE
schematics-cli: --list-schematics is broken

### ISSUE THREAD

**Issue by Splaktar** [angular/angular-cli#12220]
### Bug Report or Feature Request (mark with an `x`)
```
- [x] bug report -> please search issues before submitting
- [ ] feature request
```

### Command (mark with an `x`)
```
- [ ] new
- [ ] build
- [ ] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [ ] update
- [ ] lint
- [ ] xi18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc
- [x] schematics
```

### Versions

```
Angular CLI: 6.2.1
Node: 8.11.3
OS: darwin x64
Angular: 
... 

Package                      Version
------------------------------------------------------
@angular-devkit/architect    0.8.1
@angular-devkit/core         0.8.1
@angular-devkit/schematics   0.8.1
@schematics/angular          0.8.1
@schematics/update           0.8.1
rxjs                         6.2.2
typescript                   2.9.2
```

### Repro steps

1. `npm i -g @angular-devkit/schematics-cli @schematics/angular`
1. Verify that it installed properly via `schematics --help`

These commands give no output at all:
- `schematics @schematics/schematics --list-schematics`
- `schematics @schematics/angular --list-schematics`

### The log given by the failure

No output, even with `--verbose`.

### Desired functionality

The list of schematics in that collection as indicated by `schematics --help`:
```
schematics [CollectionName:]SchematicName [options, ...]

By default, if the collection name is not specified, use the internal collection provided
by the Schematics CLI.

Options:
    --debug             Debug mode. This is true by default if the collection is a relative
                        path (in that case, turn off with --debug=false).
    --allowPrivate      Allow private schematics to be run from the command line. Default to
                        false.
    --dry-run           Do not output anything, but instead just show what actions would be
                        performed. Default to true if debug is also true.
    --force             Force overwriting files that would otherwise be an error.
    --list-schematics   List all schematics from the collection, by name.
    --verbose           Show more information.

    --help              Show this message.

Any additional option is passed to the Schematics depending on
```
This used to work but appears to have been broken at some point. I.e. this is a regression.

**Comment by MwumLi** [comment]
ee, Has this problem been solved?

**Comment by Splaktar** [comment]
@MwumLi fixed in `7.0.0` via https://github.com/angular/angular-cli/commit/813c5209264a3c998855f7c62c5cf27321cd4ab4.
